packages/core/src/checkbox/checkbox-group.spec.ts

```diff
@@ -0,0 +1,97 @@
+import { expect, fixture } from '@open-wc/testing';
+import { sendKeys } from '@web/test-runner-commands';
+import { html } from 'lit';
+import { CheckboxGroup } from './checkbox-group.js';
+import './register.js';
+
+describe('sl-checkbox-group', () => {
+  let el: CheckboxGroup;
+
+  describe('empty', () => {
+    beforeEach(async () => {
+      el = await fixture(html`<sl-checkbox-group></sl-checkbox-group>`);
+    });
+  
+    it('should not break', () => {
+      expect(el).shadowDom.to.equalSnapshot();
+    });
+  });
+
+  describe('defaults', () => {
+    beforeEach(async () => {
+      el = await fixture(html`
+        <sl-checkbox-group>
+          <sl-checkbox>Option 1</sl-checkbox>
+          <sl-checkbox>Option 2</sl-checkbox>
+          <sl-checkbox>Option 3</sl-checkbox>
+        </sl-checkbox-group>
+      `);
+    });
+
+    it('should render correctly', () => {
+      expect(el).shadowDom.to.equalSnapshot();
+    });
+
+    it('should be valid when no option is chosen but the group is not required', () =>{
+      expect(el.internals.validity.valid).to.equal(true);
+    });
+
+    it('should handle the navigating between options correctly', async () => {            
+      expect(el.boxes?.[0].checked).to.equal(false);
+      expect(el.boxes?.[0].tabIndex).to.equal(0);
+      expect(el.boxes?.[1].checked).to.equal(false);
+      expect(el.boxes?.[1].tabIndex).to.equal(-1);
+      
+      el.boxes?.[0]?.focus();
+      await sendKeys({ press: 'Space' });
+
+      expect(el.boxes?.[0].checked).to.equal(true);
+      expect(el.boxes?.[1].checked).to.equal(false);
+      
+      await sendKeys({ press: 'ArrowRight' });
+      await sendKeys({ press: 'Enter' });
+
+      expect(el.boxes?.[0].checked).to.equal(true);
+      expect(el.boxes?.[1].checked).to.equal(true);
+    });
+
+    it('should be valid when no option is chosen but the group is not required', async () =>{
+      el.boxes?.forEach(b=>b.blur());
+      // this doesn't work, can this even be tested here?
+      // expect(el.boxes?.[0].tabIndex).to.equal(-1);
```

Comment:
Do we still want to keep these comments here?

Reply:
I don't really know how to get the test working :D Maybe Jeroen had an idea about this?

Reply:
I will take a look

Reply:
we'll pick this up in a later story